src/main/java/org/repodriller/scm/GitRepository.java

```diff
@@ -214,24 +214,35 @@ public ChangeSet getHead() {
 		List<ChangeSet> allCs = new ArrayList<>();
 
 		for (RevCommit r : git.log().all().call()) {
-			allCs.add(extractChangeSet(r));
+			allCs.add(extractChangeSet(git, r));
 		}
 		return allCs;
 	}
 
-	private ChangeSet extractChangeSet(RevCommit r) {
-		String hash = r.getName();
-		GregorianCalendar date = convertToDate(r);
+	private GregorianCalendar extractCalendar(PersonIdent pi) {
+		GregorianCalendar cal = new GregorianCalendar();
+		cal.setTimeZone(pi.getTimeZone());
+		cal.setTime(pi.getWhen());
+		return cal;
+	}
 
-		return new ChangeSet(hash, date);
+	private CommitContributor extractCommitContributor(PersonIdent pi) {
+		return new CommitContributor(pi.getName(), pi.getEmailAddress(), extractCalendar(pi), pi.getTimeZone());
 	}
 
-	private GregorianCalendar convertToDate(RevCommit revCommit) {
-		GregorianCalendar date = new GregorianCalendar();
-		date.setTimeZone(revCommit.getAuthorIdent().getTimeZone());
-		date.setTime(revCommit.getAuthorIdent().getWhen());
+	private ChangeSet extractChangeSet(Git git, RevCommit r) throws GitAPIException {
+		String id = r.getName();
+		String msg = r.getFullMessage().trim();
+		CommitContributor author = extractCommitContributor(r.getAuthorIdent());
+		CommitContributor committer = extractCommitContributor(r.getCommitterIdent());
+		List<RevCommit> parents = Arrays.asList(r.getParents());
```

Comment:
this is the point where I believe we can connect to #104. We could just load what the user really needs in her particular study.

Reply:
Agreed. I don't think we should do that in this PR though.